### DISCUSSION TITLE
Customize junitxml testcase name and classname

### DISCUSSION THREAD

**Discussion by valentin-demange** [pytest-dev/pytest#12336] [Q&A]
Hello,

An issue we have is that in the junitxml reports we have, testcase classnames and names are not convenient (see example below)
```
<testcase classname="tariffs.tests.tests_fuel_price_index_bulk_update.FuelPriceIndexBulkUpdateTransportApplicationTests" name="test_also_update_invoiced_transports_with_tariff_grid"
```

We would like to have the name to be name="{filepath}::{test_classname}::{test_name}" so that we can easily copy past to replay the failing test ?

I have try to implement with the code below, nonetheless the "::" character are always replaced in the xunitxml file. Is there a workaround ?

```
def pytest_collection_modifyitems(items):
    for item in items:
        # Split the nodeid to separate the file path from the test name
        parts = item.nodeid.split("::")
        if len(parts) > 2:
            # Reconstruct the nodeid to include the class name for test methods
            class_name = parts[-2]
            test_name = parts[-1]
            file_path = parts[0]
            item._nodeid = f"{file_path}::{class_name}::{test_name}"
```

**Comment by nicoddemus** [comment]
I don't know about an easy way to do this, but an alternative is to post-process the XML file and change the ids as you prefer.

**Reply by valentin-demange** [reply]
No convenient, the "::' syntax is on one side used by pytest to run the tests, but on the other side not accepted in the test report (junitxml format) as test name.

Thanks for your quick answer btw
